filesystems/tests/common.py

```diff
@@ -1092,3 +1137,23 @@ def test_readlink_relative(self):
         fs.link(source=source, to=to)
 
         self.assertEqual(fs.readlink(to), source)
+
+    def _invalid_mode(self, mode):
+        fs = self.FS()
+        tempdir = fs.temporary_directory()
+        self.addCleanup(fs.remove, tempdir)
+
+        def fail():
+            with fs.open(tempdir.descendant("unittesting", mode)):
+                pass
+
+        self.assertRaises(exceptions.InvalidMode, f=fail)
+
+    def test_invalid_mode_activity(self):
+        self._invalid_mode('z')
+
+    def test_invalid_mode_mode(self):
+        self._invalid_mode('rz')
+
+    def test_invalid_mode_extra(self):
+        self._invalid_mode('rbz')
```

Comment:
Even a copy of these tests with `'rb'` passes...  time to correct that.

Reply:
Hmm, it seems that `f=fail` doesn't work despite the docs:
https://twistedmatrix.com/documents/current/api/twisted.trial._synctest._Assertions.html#assertRaises
```python
def assertRaises(self, exception, *args, f=None, **kwargs):
```
I had to switch to passing it as a positional arg to get `fail()` to actually be called.

Reply:
So... yeah.  I saw the tests were being run by Trial...  and didn't check to see what function I was really calling.  This is actually https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises.  Interesting effect here when I accidentally used the context manager version...  of to switch to that though.